CodeModel/CodeParsers/CSharp/ProjectAnalyzer/CohesiveParts.cs

```diff
@@ -0,0 +1,16 @@
+﻿using System.Collections.Generic;
+
+namespace CodeModel.CodeParsers.CSharp.ProjectAnalyzer
+{
+    public class CohesiveParts
+    {
+        public List<HashSet<Access>> Parts { get; }
```

Comment:
Refactoring proposition: create a wrapper class around `HashSet<Access>`. Many methods have this collection as a parameter. Also by doing it I would avoid using `List<HashSet<Access>>`, and have a more meaningful name.
I am not sure about the class name. I was thinking about `Part` since it represents a part of a class, could be whole class, could be few method-field interactions. Maybe `ClassPart` since `Part` is too generic?